sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/WebPubSubServiceClientTests.java

```diff
@@ -87,6 +87,14 @@ public void testBroadcastString() {
                         .set("Content-Type", "text/plain"))), 202);
     }
 
+    @Test
+    public void testBroadcastStringWithFilter() {
+        assertResponse(client.sendToAllWithResponse(
+            BinaryData.fromString("Hello World - Broadcast test!"),
+            new RequestOptions().setHeader("Content-Type", "text/plain")
+                .addQueryParam("filter", "userId ne 'user1'")), 202);
```

Comment:
How about adding a ClientConnectionFilter helper class, as similar to what azure search does with its [SearchFilter](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchFilter.java#L35)

A similar approach in c# is here https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/webpubsub/Azure.Messaging.WebPubSub/src/ClientConnectionFilter.cs